src/Illuminate/Database/Connectors/SqlServerConnector.php

```diff
@@ -45,12 +45,33 @@ protected function getDsn(array $config)
         // need to establish the PDO connections and return them back for use.
         if (in_array('dblib', $this->getAvailableDrivers())) {
             return $this->getDblibDsn($config);
+        } elseif (in_array('odbc', $this->getAvailableDrivers())) {
+            if (isset($config['driver']) && $config['driver'] === 'odbc') {
+                return $this->getOdbcDsn($config);
+            } else {
+                return $this->getSqlSrvDsn($config);
```

Comment:
What if odbc is an availble driver but they want to use the sqlsrv string? By taylorotwell.